examples/threading/threading.c

```diff
@@ -13,13 +16,40 @@ void* threadfunc(void* thread_param)
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
-    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+
+    usleep(thread_func_args->wait_to_obtain_ms * MILLS_TO_MICROS);
```

Comment:
@dhirajbennadi can you be sure the usleep completed it's wait here?  Or do you need a return code check?